utils/build-script-impl

```diff
@@ -1636,23 +1638,31 @@ for host in "${ALL_HOSTS[@]}"; do
                 ;;
 
             llvm)
-                if [ "${BUILD_LLVM}" == "0" ] ; then
-                    build_targets=(clean)
-                fi
-                if [[ "${SKIP_BUILD}" || "${SKIP_BUILD_LLVM}" ]] ; then
-                    # We can't skip the build completely because the standalone
-                    # build of Swift depend on these for building and testing.
-                    build_targets=(llvm-tblgen clang-resource-headers intrinsics_gen clang-tablegen-targets)
-                    # If we are not performing a toolchain only build, then we
-                    # also want to include FileCheck, not, llvm-nm for testing
-                    # purposes.
-                    if [[ ! "${BUILD_TOOLCHAIN_ONLY}" ]] ; then
-                      build_targets=(
-                          "${build_targets[@]}"
-                          FileCheck
-                          not
-                          llvm-nm
-                      )
+                if [ "${LLVM_NINJA_TARGETS}" != "" ] ; then
+                    if [[ $(is_cross_tools_host ${host}) && "${LLVM_NINJA_TARGETS_FOR_CROSS_COMPILE_HOSTS}" != "" ]] ; then
+                        build_targets=("${LLVM_NINJA_TARGETS_FOR_CROSS_COMPILE_HOSTS[@]}")
+                    else
+                        build_targets=("${LLVM_NINJA_TARGETS[@]}")
+                    fi
```

Comment:
I think that adding the additional check would make this clearer:

```
if [[ -n "${LLVM_NINJA_TARGETS_FOR_CROSS_COMPILE_HOSTS} && $(is_cross_tools_host ${host}) ]] ; then
  build_targest=( "${LLVM_NINJA_TARGETS_FOR_CROSS_COMPILE_HOSTS[@]}" )
elif [[ -n "${LLVM_NINJA_TARGETS}" ]] ; then
  build_targest=( "${LLVM_NINJA_TARGETS[@]}" )
else
  ...
fi
```

Reply:
Makes sense, that additional check does not bring any value.

Amended in e98ce3db4eb